src/components/ImportForm/SourceSection/SourceSection.tsx

```diff
@@ -91,7 +94,9 @@ const SourceSection: React.FC<SourceSectionProps> = () => {
     if (accessCheckLoaded) {
       if (isRepoAccessible) {
         try {
-          const { organization } = gitUrlParse(sourceUrl);
+          const { organization, name } = gitUrlParse(sourceUrl);
+          !inAppContext && setFieldValue('application', incrementNameCount(applications, name));
```

Comment:
Do we actually need to do this in SourceSection? Maybe just update the application name in `ApplicationSection`. The hook `useApplicationName` can take a Git URL and return back a valid application based on git repo.

Reply:
I think we can also update it for sample import flow this way

Reply:
Moved the hook usage to review section. For samples flow, I am retaining the same recommendation logic as before.